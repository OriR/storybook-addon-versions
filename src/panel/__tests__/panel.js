import React from 'react';
import renderer from 'react-test-renderer';
import Panel from '../';

jest.mock('../../utils/config');

test('Panel renders correctly, dev false', () => {
  const storybook = {
    getQueryParam: () => 'false',
    setQueryParams: () => {},
  };

  const tree = renderer
    .create(<Panel storybook={storybook} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Panel renders correctly, dev true', () => {
  const storybook = {
    getQueryParam: () => 'true',
    setQueryParams: () => {},
  };

  const tree = renderer
    .create(<Panel storybook={storybook} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
