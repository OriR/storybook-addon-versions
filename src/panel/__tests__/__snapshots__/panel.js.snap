// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly, dev false 1`] = `
<div
  className="versions-panel-container"
>
  <input
    checked={false}
    id="versionsAddonDevMode"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="versionsAddonDevMode"
  >
    Developer mode
  </label>
  <div
    className="versions-panel-list"
  >
    <p>
      No versions found
    </p>
  </div>
</div>
`;

exports[`Panel renders correctly, dev true 1`] = `
<div
  className="versions-panel-container"
>
  <input
    checked={true}
    id="versionsAddonDevMode"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="versionsAddonDevMode"
  >
    Developer mode
  </label>
  <div
    className="versions-panel-list"
  >
    <p>
      No versions found
    </p>
  </div>
</div>
`;
